app/registries/foreman/plugin.rb

```diff
@@ -317,6 +317,24 @@ def tests_to_skip(hash)
       end
     end
 
+    # Adds setting definition
+    #
+    # ===== Example
+    #
+    #   settings do |mapper|
+    #     mapper.category(:cfgmgmt, N_('Configuration Management')) do |cat_mapper|
+    #       cat_mapper.setting(:use_cooler_puppet, default: true,
```

Comment:
Have you considered going a step further in the DSL and support
```suggestion
    #   settings do
    #     category(:cfgmgmt, N_('Configuration Management')) do
    #       setting(:use_cooler_puppet, default: true,
```

According to http://rubylearning.com/blog/2010/11/30/how-do-i-build-dsls-with-yield-and-instance_eval/ the trick is to use `instance_eval &block`.

I do believe it'll end up with a nicer to read DSL.

Reply:
Great idea :+1:
I'm definitelly up for that, I'd even not go with the `mapper.setting` support at all and support just the format you're proposing! :)